pkg/didcomm/protocol/issuecredential/service.go

```diff
@@ -346,12 +346,7 @@ func nextState(msg service.DIDCommMsg, outbound bool) (state, error) {
 
 // canTriggerActionEvents checks if the incoming message can trigger an action event
 func canTriggerActionEvents(msg service.DIDCommMsg) bool {
-	switch msg.Type() {
-	case ProposeCredentialMsgType, OfferCredentialMsgType, RequestCredentialMsgType, IssueCredentialMsgType:
-		return true
-	}
-
-	return false
+	return msg.Type() == ProposeCredentialMsgType || msg.Type() == OfferCredentialMsgType
```

Comment:
Don't we need user interaction/handling to pass the actual Credentials when `request credential` message type is received ?

Reply:
It will be done by passing an option to the continue function or executing a specific method from the client. This PR is about `states transitions`. So, I have described the transitions between states.

Reply:
Yes, but for that we need to trigger events right ? I see we are removing it in this PR.

Reply:
I am going to add it

Reply:
Added `RequestCredentialMsgType`